src/fastcs/connections/serial_connection.py

```diff
@@ -0,0 +1,54 @@
+import asyncio
+from dataclasses import dataclass
+
+import aioserial
+
+
+class NotOpenedError(Exception):
+    pass
+
+
+@dataclass
+class SerialConnectionSettings:
+    port: str
+    baud: int = 115200
+
+
+class SerialConnection:
+    def __init__(self):
+        self.aioserial_instance = None
+        self._lock = asyncio.Lock()
+
+    async def connect(self, settings: SerialConnectionSettings) -> None:
+        self.aioserial_instance = aioserial.AioSerial(
```

Comment:
I see this is the name used in the aioserial examples, but it isn't very clear. Could we use something like stream, port, socket, device?